hazelcast-jet-sql/src/test/java/com/hazelcast/jet/sql/impl/connector/test/TestEmptyStreamSqlConnector.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.sql.impl.connector.test;
+
+import com.hazelcast.jet.core.DAG;
+import com.hazelcast.jet.core.TestProcessors.MockP;
+import com.hazelcast.jet.core.Vertex;
+import com.hazelcast.jet.sql.impl.connector.SqlConnector;
+import com.hazelcast.jet.sql.impl.schema.JetTable;
+import com.hazelcast.jet.sql.impl.schema.MappingField;
+import com.hazelcast.spi.impl.NodeEngine;
+import com.hazelcast.sql.impl.QueryException;
+import com.hazelcast.sql.impl.expression.Expression;
+import com.hazelcast.sql.impl.schema.ConstantTableStatistics;
+import com.hazelcast.sql.impl.schema.Table;
+import com.hazelcast.sql.impl.schema.TableField;
+import com.hazelcast.sql.impl.type.QueryDataType;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Map;
+
+import static com.hazelcast.jet.impl.util.Util.toList;
+import static java.util.Collections.singletonList;
+
+/**
+ * A connector for SQL that has a single column `v` of type BIGINT, is a
+ * streaming type, but never produces any items..
+ */
+public class TestEmptyStreamSqlConnector implements SqlConnector {
```

Comment:
Can't we use simply `TestBatchSqlConnector` with zero items?

Reply:
Nope, that would complete. I want it to get stuck waiting for more items.